ballerina/googlesheets_client.bal

```diff
@@ -0,0 +1,520 @@
+    # + fields - The fields to be retrieved
+    # + include - The relations to be retrieved (SQL `JOINs` to be performed)
+    # + typeDescriptions - The type descriptions of the relations to be retrieved
+    # + return - A record in the `rowType` type or a `persist:Error` if the operation fails
+    public isolated function runReadByKeyQuery(typedesc<record {}> rowType, typedesc<record {}> rowTypeWithIdFields, map<anydata> typeMap, anydata key, string[] fields = [], string[] include = [], typedesc<record {}>[] typeDescriptions = []) returns record {}|Error {
+        record {} 'object = check self.queryOne(key);
+
+        'object = filterRecord('object, self.addKeyFields(fields));
+        check self.getManyRelations('object, fields, include, typeDescriptions);
+        self.removeUnwantedFields('object, fields);
+
+        do {
+            return check 'object.cloneWithType(rowType);
+        } on fail error e {
+            return <Error>e;
+        }
+    }
+
+    # + rowType - The type description of the entity to be retrieved
+    # + typeMap - The data types of the record
+    # + fields - The fields to be retrieved
+    # + include - The associations to be retrieved
+    # + return - A stream of records in the `rowType` type or a `persist:Error` if the operation fails
+    public isolated function runReadQuery(typedesc<record {}> rowType, map<anydata> typeMap, string[] fields = [], string[] include = [])
+    returns stream<record {}, error?>|Error {
+        return self.query(self.addKeyFields(fields));
+    }
+
+    # + rowType - The type description of the entity to be retrieved
+    # + typeMap - The data types of the record
+    # + fields - The fields to be retrieved
+    # + include - The associations to be retrieved
+    # + return - A stream of records in the `rowType` type or a `persist:Error` if the operation fails
+    public isolated function readTableAsStream(typedesc<record {}> rowType, map<anydata> typeMap, string[] fields = [], string[] include = []) returns stream<record {}, Error?>|Error {
+        string query = "select *";
+        string|error encodedQuery = url:encode(query, "UTF-8");
+        if encodedQuery is error {
+            return <Error>error(encodedQuery.message());
+        }
+        http:QueryParams queries = {"gid": self.sheetId, "range": self.range, "tq": <string>encodedQuery, "tqx": "out:csv"};
+        http:Response|error response = self.httpClient->/d/[self.spreadsheetId]/gviz/tq(params = queries);
+        if response is error {
+            return <Error>error(response.message());
+        }
+        string|error textResponse = response.getTextPayload();
+        if (textResponse !is error) {
+            string[] responseRows = re `\n`.split(textResponse);
+            record {}[] rowTable = [];
+            if responseRows.length() == 0 {
+                return <Error>error("Error: the spreadsheet is not initialised correctly. ");
```

Comment:
```suggestion
                return <Error>error("Error: the spreadsheet is not initialised correctly.");
```

Reply:
updated in [4346c39](https://github.com/ballerina-platform/module-ballerina-persist/pull/201/commits/4346c3959a465aeb05de0872b94ea1056a715a9a)